localstack/http/hypercorn.py

```diff
@@ -82,15 +82,19 @@ class GatewayServer(HypercornServer):
     """
 
     def __init__(
-        self, gateway: Gateway, listen: HostAndPort | list[HostAndPort], use_ssl: bool = False
+        self,
+        gateway: Gateway,
+        listen: HostAndPort | list[HostAndPort],
+        use_ssl: bool = False,
+        threads: int | None = None,
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (performance):** Consider providing a default value for `threads` that aligns with system capabilities.

Using `None` as a default for `threads` and later substituting it with a hardcoded value (1000) might not be optimal for all environments. Consider a default that scales with the number of available CPU cores.

```suggestion
# Define a constant for the default gateway worker count
DEFAULT_GATEWAY_WORKER_COUNT = 1000

GATEWAY_WORKER_COUNT = int(os.environ.get("GATEWAY_WORKER_COUNT", DEFAULT_GATEWAY_WORKER_COUNT))
```